src/v/storage/segment.cc

```diff
@@ -583,7 +583,7 @@ ss::future<append_result> segment::do_append(const model::record_batch& b) {
 }
 
 ss::future<append_result> segment::append(const model::record_batch& b) {
-    if (has_compaction_index() && b.header().attrs.is_transactional()) {
+    if (has_compaction_index() && b.contains_transactional_data()) {
```

Comment:
Nice! This isn't specific to windowed compaction, is it? Should it be backported?

Reply:
right.. I realized this while debugging something else.